### PR TITLE
added failing test for #13522

### PR DESCRIPTION
**Pull request by cebe**
added failing test for #13522

regression after 2.0.11 release.

~~Working on a fix.~~ - done.

### CODE REVIEW

**Comment by cebe** [comment]
@rob006 as you have been involved in UrlManager changes in the past, you might want to take a look at this one.

**Comment by rob006** [comment]
I think that #12691 created this bug and this should be fixed in `UrlRule`.

**Comment by cebe** [comment]
I do not see how exactly it introduced it but I agree that it may be better to fix it in UrlRule.

**Comment by rob006** [comment]
I checked again and I think I'm wrong. This is not regression, this bug exist even in 2.0.5. In that case this fix looks valid to me.

File: framework/web/UrlManager.php [hunk 1]
```diff
@@ -414,6 +415,7 @@ public function createUrl($params)
                 $route .= '?' . $query;
             }
 
+            $route = ltrim($route, '/');
```
Comment (rob006) [inline framework/web/UrlManager.php]: This is not needed, `$route` is already trimmed: https://github.com/cebe/yii2/blob/9a53571fd2cac0155477bef8420a74b0b665f2c7/framework/web/UrlManager.php#L360

Comment (cebe) [inline framework/web/UrlManager.php]: this fixes the case where route is empty and a `/` suffix is used. see the test case added below (line 447).

**Review by samdark — APPROVED** [review]

**Review by rob006 — APPROVED** [review]

**Review by SilverFire — APPROVED** [review]

**Comment by samdark** [comment]
@cebe merge?

**Pull request closed without merging**